### PR TITLE
#169817894:A manager should get notification of edited trip requests from their direct reports

### PR DESCRIPTION
**Pull request by mujohn26**
#### What does this PR do?

This PR adds notification functionality.
When the requester edit a trip a  notification will be sent to the manager for both in-app and via email.
#### Description of Task to be completed?

* There may be 2 types of notifications: email notifications and in-app notifications.
* Email notifications are sent as email messages to the email address the user submitted during registration.
* Users should be able to opt out (and in) of this type of notification.
* In-app notifications show up in the app. This may be visible on a notification page or some other view, as the team may choose.
#### How should this be manually tested?

- After clone repository, go to the root of the project open path into command run the `npm install` and once setup is done click `npm test`.

- In Postman, create a GET request with this url:`localhost:3000/api/v1/notifications` and pass `token` in headers
#### Screenshots

Screen Shot 2020-01-17 at 12 51 54

#### What are the relevant pivotal tracker stories?
[#169817894](https://www.pivotaltracker.com/story/show/169817894)

### CODE REVIEW

**Review by VedaDom — APPROVED** [review]

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -91,6 +91,7 @@ class userController {
     if (updaUser.status === 200) {
```
Comment (jo-rdan) [inline src/controllers/user.controller.js]: @mujohn26 , this variable seems to be misspelled, kindly update this to a more intuitive name

**Review by jo-rdan — CHANGES_REQUESTED** [review]
Good job so far @mujohn26, one little thing to change and I am ok!

File: src/controllers/trip.controller.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ class TripController {
       };
       const tripRequest = await tripService.CreateTripRequest(requestData);
       await NotificationService.sendNotification('trip_request_event', managerId, `New ${type} trip request.`, `${req.user.firstName} has requested a new ${type}`, tripRequest.id, 'http://localhost:3000/api/v1/notifications');
```
Comment (bdushimi) [inline src/controllers/trip.controller.js]: @mujohn26 Kindly use the environment variable to point to a particular URL depending on the application environment

File: src/controllers/trip.controller.js [hunk 3]
```diff
@@ -216,10 +216,96 @@ class TripController {
    */
   static async getTripRequest(req, res) {
     const subjectID = req.params.tripRequestID;
-    const userID = req.user.id;
-    const data = await tripService.getTripRequest(subjectID, userID);
+    const data = await tripService.getTripRequest(subjectID);
     if (data) return response.successMessage(res, 'success', 200, data);
     return response.errorMessage(res, 'Trip not found', 404);
   }
+
```
Comment (bdushimi) [inline src/controllers/trip.controller.js]: @mujohn26 This PR is introducing changes that are not relevant to its purpose. The PR contains codes related to updating trips and trip-requests yet this PR is supposed to contain codes that send notifications to a requester's manager. I think you branched out of edit-trip-request branch. Before you start off a new feature, always branch out from develop.

**Review by bdushimi — CHANGES_REQUESTED** [review]

**Pull request merged**